CMakeLists.txt

```diff
@@ -78,6 +78,14 @@ if(NOT MSVC)
   set(CMAKE_C_STANDARD_REQUIRED True)
 endif()
 
+# Option for packagers to add a suffix after "adios" in library names (to
+# install multiple builds in parallel)
+set(ADIOS2_LIBRARY_SUFFIX "" CACHE STRING
+  "Suffix to put after \"adios\" in library names")
+set(ADIOS2_EXECUTABLE_SUFFIX "" CACHE STRING
```

Comment:
How would these two be different?  Just curious as to what the actual use case is.

Reply:
`ADIOS2_LIBRARY_SUFFIX` would be `_serial` or `_openmpi` (resulting in `libadios2_serial_core.so`) while `ADIOS2_EXECUTABLE_SUFFIX` would be `.serial` or `.openmpi` (resulting in `bpls.serial`.)

Reply:
Is there really a need for two separate suffixes that will generally be almost the same?  It seems like just one applying the same to both libraries and executables should be sufficient.

Reply:
The difference is that one uses a `_` while the other uses a `.`. We could have some sort of logic that consolidates the two and only includes the `_` or `.` if the suffix is non-empty.

Reply:
Sure, but why make them different at all?  The whole point is to just have them distinctly tagged for simultaneous installs

Reply:
This is based on the convention that HDF5 uses in Debian. It uses `libhdf5_openmpi_foo.so` and `h5cc.openmpi`.

Reply:
Right, but we should be able to do all the necessary separation and symlinking with `libadios2_core.so -> libadios2_openmpi_core.so ` and `bpls -> bpls_openmpi`.  Even if it's different from how the `hdf5` deb package is setup it still achieves the necessary capability to do what we need to do in the package, right?

Reply:
It does, but I worry that I may be asked to change it to match the HDF5 convention.

As a middle ground, what if we made `ADIOS2_EXECUTABLE_SUFFIX` default to the same value as `ADIOS2_LIBRARY_SUFFIX`, so that if we don't need to make them different, we can just set the one?

Reply:
That's acceptable then I think.